### DISCUSSION TITLE
Support between for every

### DISCUSSION THREAD

**Discussion by jimtng** [boc-tothefuture/openhab-jruby#140] [Ideas]
Currently, `every` only supports `at`. It would be great to be able to also support `between` and `on`

```
rule 'test' do
  every 30.minutes, between: '8am'..'2pm'
# translates to 0 */30 8-14 ? * *

  every :hour, between: '8am'..'2pm'
# 0 0 8-14 ? * *

  every :day, between: '8am'..'2pm' # the same?
# 0 0 8-14 ? * *

  every :day, between: '8am'...'2pm' 
# 0 0 8-13 ? * *

  every :week, between: '8am'..'2pm'
# 0 0 8-14 ? * MON

  every :week, on: TUE, between: '8am'..'2pm'
# 0 0 8-14 ? * TUE

  every TUE, between: '8am'..'2pm'
# 0 0 8-14 ? * TUE # the same

  every :week, on: [TUE, THU, SAT], between: '8am'..'2pm'
# 0 0 8-14 ? * TUE,THU,SAT

  every [TUE, THU, SAT], between: '8am'..'2pm'
# 0 0 8-14 ? * TUE,THU,SAT # the same

  every :month, between: '8am'..'2pm'
# 0 0 8-14 ? 1 ?

  every :month, on: 3, between: '8am'..'2pm'
# 0 0 8-14 ? 3 ?

  every 2.hours, between: '8am'..'2pm'
# 0 0 8,10,12,14 ? * * - this is trickier

  every 2.hours, between: '8am'...'2pm'
# 0 0 8,10,12 ? * * 

end

```

**Comment by boc-tothefuture** [comment]
We already support the days of the week.
```every :tuesday ```

**Comment by boc-tothefuture** [comment]
We also have between as a guard..  It might be slightly less efficient, but the dispatch and guard should be really really quick unless we can prove otherwise.

If we had the between with dates like we talked about it would be duplicated again.

Another option would be if we have a between in a guard and it is an every if it's possible we automatically use it to update the cron job. That way it keeps the guard concept, is optimized and invisible to the user.

**Comment by jimtng** [comment]
> Another option would be if we have a between in a guard and it is an every if it's possible we automatically use it to update the cron job. That way it keeps the guard concept, is optimized and invisible to the user.
> 

I like this option
